e2e/benchmarks/local-benchmark/dump.js

```diff
@@ -0,0 +1,266 @@
+  }
+
+  return tensorMap;
+}
+
+async function predictAndGetData(
+    predict, model, inferenceInput, benchmark, enableDump) {
+  const prediction = await predict(model, inferenceInput);
+  const graphModel = getGraphModel(model);
+  let intermediateData = {};
+  enableDump = enableDump && !!graphModel;
+  if (enableDump) {
+    intermediateData =
+        await getIntermediateTensorInfo(graphModel.getIntermediateTensors());
+  }
+  const predictionData = await getPredictionData(prediction);
+  if (enableDump) {
+    graphModel.disposeIntermediateTensors();
+  }
+  return {data: predictionData, intermediateData};
+}
+
+async function predictOp(
+    model, modelJson, dumpedJson, outputNodeName, backend) {
+  await tf.setBackend(backend);
+  const tensorMap =
+      await createNamedTensorMap(outputNodeName, modelJson, dumpedJson);
+  if (tensorMap == null) {
+    return null;
+  }
+  let prediction;
+  let savedKeepIntermediateTensors;
+  try {
+    savedKeepIntermediateTensors =
+        tf.env().getBool('KEEP_INTERMEDIATE_TENSORS');
+    tf.env().set('KEEP_INTERMEDIATE_TENSORS', false);
+  } catch (e) {
+    console.warn(e.message);
+  }
+  try {
+    // TODO(#6861): Support tensor with type conversion.
+    prediction = await model.executeAsync(tensorMap, outputNodeName);
+  } catch (e) {
+    tf.env().set('KEEP_INTERMEDIATE_TENSORS', savedKeepIntermediateTensors);
+    console.warn(e.message);
+    return null;
+  }
+
+  const predictObject = await getPredictionData(prediction, true);
+  tf.env().set('KEEP_INTERMEDIATE_TENSORS', savedKeepIntermediateTensors);
+  return predictObject;
+}
+
+/**
+ * Dump the predict results of two backends and save diffs to files.
+ * @param model The loaded model.
+ * @param input The predict results from backends.
+ * @param prefix Used for generating dump file name.
+ * @param level 0, dump big diffs. 1, dump any diffs.
+ * @param length Used for controlling how many tensors will be dumped. -1 dump
+ *     all.
+ */
+async function dump(
+    model, input, prefix = '', level = DUMP_LEVEL.BIGDIFF, length = 1) {
+  const graphModel = getGraphModel(model);
+  if (graphModel == null) {
+    return;
+  }
+  const backends = Object.keys(input);
+  const jsonObject1 = input[backends[0]];
+  const jsonObject2 = input[backends[1]];
+  const dumpObjects1 = {};
+  const dumpObjects2 = {};
+  const keys = Object.keys(jsonObject1);
+  prefix = `dump_${prefix}_${level}`;
+  let dumpCount = 0;
+
+  const modelJson = graphModel.artifacts;
+  for (let i = 0; i < keys.length; i++) {
+    const key = keys[i];
+    if (compareData(jsonObject1[key], jsonObject2[key], level)) {
+      continue;
+    }
+    const predictObject =
+        await predictOp(graphModel, modelJson, jsonObject2, key, backends[0]);
+    const [objects1, objects2] = predictObject ?
+        [{...predictObject, i}, {...jsonObject2[key], i}] :
+        [null, null];
+    if (objects1 && objects2 && !compareData(objects1, objects2, level)) {
```

Comment:
Similar as above, we need an early continue here if the data are same.

Reply:
Done